libmatrix.c

```diff
@@ -54,6 +55,33 @@ static const char *matrixprpl_list_icon(PurpleAccount *acct, PurpleBuddy *buddy)
 }
 
 /**
+ * Callback for when joining a room fails, create the room
+ */
+static void room_create_callback(MatrixConnectionData *conn,
+        gpointer user_data, int http_response_code,
+        struct _JsonNode *json_root)
+{
+    const gchar *errorstr, *errcodestr;
+    JsonObject *root_obj;
+    const gchar *room_alias;
+
+    if (json_root) {
+        root_obj = matrix_json_node_get_object(json_root);
+    }
+    errorstr = json_object_get_string_member(root_obj, "error");
+    errcodestr = json_object_get_string_member(root_obj, "errcode");
+    char roomid[strlen(errorstr) - 20];
+    if (errcodestr && !strcmp(errcodestr, "M_NOT_FOUND")) {
+        snprintf(roomid, strlen(errorstr) - 20, "%s", &errorstr[11]);
```

Comment:
is this trying to fish the room alias out of the error string? what happens if the error string changes?

You should be able to get the alias from the user_data.